### PR TITLE
Include pdbs with nuget and zip packges and publish source code

### LINKED ISSUE #551: Make engine easier to Debug
I'm phrasing this as a set of questions. Depending on the answers, it might become a task.

When using the engine as a component of another runner via nuget, debugging is rather tricky.

1. Is there a reason that the pdb files are not distributed with the nuget packages? That's done with the adapter, for example.

2. If I build for debug, the version suffix changes in the assemblyinfo file. I'm wondering if use of "-dbg" at the end of the suffix is something folks are attached to or which seems like it would break anyone if we did it differently. I see a couple of alternatives
 2.1 Add -dbg or (Debug) to the name component name, making it a different package
 2.2 Don't change the name at all but just create a separate feed for debug builds.

3. How do you feel about having separate build targets for the engine and console?

These questions could apply to the framework as well, but it's the engine that I'm trying to debug under the GUI right now.

### PR DESCRIPTION
**Pull request by CharliePoole**
Fixes #551

Everything is tested except for publishing source to myget. We can only see that work after merging.

### CODE REVIEW

File: package-checks.cake [hunk 1]
```diff
@@ -58,18 +92,16 @@ public void CheckAllPackages()
...
+            HasDirectory("bin/netstandard2.0").WithFiles(ENGINE_FILES).AndFiles(ENGINE_PDB_FILES),
+            HasDirectory("bin/netcoreapp2.1").WithFiles(ENGINE_FILES).AndFiles(ENGINE_PDB_FILES),
+            HasDirectory("bin/netcoreapp3.1").WithFiles(ENGINE_FILES).AndFiles(ENGINE_PDB_FILES),
+            HasDirectory("bin/agents/net20").WithFiles(AGENT_FILES).AndFiles(AGENT_PDB_FILES),
+            HasDirectory("bin/agents/net40").WithFiles(AGENT_FILES).AndFiles(AGENT_PDB_FILES)) &
         CheckMsiPackage("NUnit.Console", 
             HasDirectory("NUnit.org").WithFiles("LICENSE.txt", "NOTICES.txt", "nunit.ico"),
-            HasDirectory("NUnit.org/nunit-console").WithFiles("nunit3-console.exe", "nunit3-console.exe.config").AndFiles(ENGINE_FILES).AndFile("nunit.bundle.addins"),
-            HasDirectory("NUnit.org/nunit-console/agents/net20").WithFiles("nunit-agent.exe", "nunit-agent.exe.config", "nunit-agent-x86.exe", "nunit-agent-x86.exe.config", "nunit.agent.addins"),
-            HasDirectory("NUnit.org/nunit-console/agents/net40").WithFiles("nunit-agent.exe", "nunit-agent.exe.config", "nunit-agent-x86.exe", "nunit-agent-x86.exe.config", "nunit.agent.addins"),
```
Comment (mikkelbu) [inline package-checks.cake:72]: It looks like these two have disappeared. Is that on purpose?

**Review by mikkelbu — COMMENTED** [review]
A couple of comments/questions

**Review by rprouse — APPROVED** [review]
Looks good to me except for the two lines that @mikkelbu mentioned.

**Comment by CharliePoole** [comment]
It has been a long time since I submitted this but IIRC those two lines failed. Some checks couldn't be done on the msi because I used an administrative install to do the checking. That expands the msi but doesn't put everything in its final location. Using a normal install might be possible if somebody wants to try it but it requires permissions the test runner doesn't normally have. I think I just decided to skip that particular check as an easy out.

**Comment by mikkelbu** [comment]
> It has been a long time since I submitted this but IIRC those two lines failed. Some checks couldn't be done on the msi because I used an administrative install to do the checking. That expands the msi but doesn't put everything in its final location. Using a normal install might be possible if somebody wants to try it but it requires permissions the test runner doesn't normally have. I think I just decided to skip that particular check as an easy out.

I can live with leaving these lines out. I'll merge it now and then we can always improve on this, if we feel for it later on.

File: build.cake [hunk 2]
```diff
@@ -380,61 +380,38 @@ Task("BuildNuGetPackages")
...
         {
             Version = productVersion,
             BasePath = CURRENT_IMG_DIR,
             OutputDirectory = PACKAGE_DIR,
-            NoPackageAnalysis = true
-        });
+            NoPackageAnalysis = true,
+            Symbols = true,
+            // Not yet supported by cake as a setting
+            ArgumentCustomization = args => args.Append("-SymbolPackageFormat snupkg")
```
[the remark this answers is not in the document]
Comment (mikkelbu) [reply build.cake:399]: I forgot about this, when I merged. I've created a new issue for this (in #1029)

**Pull request merged**